pinkopy/base_session.py

```diff
@@ -96,101 +97,99 @@ def __enter__(self):
     def __exit__(self, exception_type, exception_value, traceback):
         self.logout()
 
-    def request(self, method, path, attempt=None, headers=None, payload=None,
-                payload_nondict=None, qstr_vals=None, service=None):
-        """Make request.
+    def request(self, method, path, attempt=None, headers=None, payload=None, payload_nondict=None, qstr_vals=None, service=None):
+        """
+        Make an HTTP request.
 
         Args:
-            method (str): HTTP method
-            path (str): request path
-            attempt (int): Number of request attempts
-            headers (optional[dict]): headers if provided else self.headers
-            payload (optional[dict]): payload as dictionary
-            payload_nondict (optional[str]): payload raw data
-            qstr_vals (optional[dict]): query string parameters to add
-            service (optional[str]): URL and path to root of api
+            method (str): HTTP method.
+            path (str): Request path.
+            attempt (int, optional): Number of request attempts.
+            headers (dict, optional): Request headers. Defaults to self.headers.
+            payload (dict, optional): Request payload as a dictionary.
+            payload_nondict (str, optional): Request raw data payload.
+            qstr_vals (dict, optional): Query string parameters to add.
+            service (str, optional): URL and path to root of API. Defaults to self.service.
 
         Returns:
-            response object
+            requests.Response: Response object.
+
+        Raises:
+            PinkopyError: If an error occurs during the request.
         """
-        # We may need to recall the same request.
-        # Must pop self because it is passed implicitly and cannot be passed twice.
-        _context = {k: v for k, v in locals().items() if k is not 'self'}
+        _context = {k: v for k, v in locals().items() if k != 'self'}
         allowed_attempts = 3
-        attempt = 1 if not attempt else attempt
-        service = service if service else self.service
-        headers = headers if headers else self.headers
+        attempt = attempt or 1
+        service = service or self.service
+        headers = headers or self.headers
```

Comment:
I don't think it is an improvement to short circuit this way, and is generally more preferred to use the removed structure, so it seems like more preferences.